packages/fiori/src/Wizard.ts

```diff
@@ -50,105 +58,34 @@ const STEP_SWITCH_THRESHOLDS = {
 	MAX: 1,
 };
 
-const RESPONSIVE_BREAKPOINTS = {
+type ResponsiveBreakpoints = {
+	[key: string]: string,
+}
+
+const RESPONSIVE_BREAKPOINTS: ResponsiveBreakpoints = {
 	"0": "S",
 	"599": "M",
 	"1023": "L",
 	"1439": "XL",
 };
 
 /**
+ * Fired when the step is changed by user interaction - either with scrolling,
+ * or by clicking on the steps within the component header.
+ *
+ * @event sap.ui.webc.fiori.Wizard#step-change
+ * @param {WizardStep} step The new step.
+ * @param {WizardStep} previousStep The previous step.
+ * @param {boolean} changeWithClick The step change occurs due to user's click or 'Enter'/'Space' key press on step within the navigation.
  * @public
  */
-const metadata = {
-	tag: "ui5-wizard",
-	managedSlots: true,
-	fastNavigation: true,
-	properties: /** @lends sap.ui.webc.fiori.Wizard.prototype */ {
-		/**
-		 * Defines the width of the <code>ui5-wizard</code>.
-		 * @private
-		 */
-		width: {
-			type: Float,
-		},
-
-		/**
-		 * Defines the threshold to switch between steps upon user scrolling.
-		 * <br><br>
-		 *
-		 * <b>For Example:</b>
-		 * <br>
-		 * (1) To switch to the next step, when half of the step is scrolled out - set <code>step-switch-threshold="0.5"</code>.
-		 * (2) To switch to the next step, when the entire current step is scrolled out - set <code>step-switch-threshold="1"</code>.
-		 *
-		 * <br><br>
-		 * <b>Note:</b> Supported values are between 0.5 and 1
-		 * and values out of the range will be normalized to 0.5 and 1 respectively.
-		 * @private
-		 * @type {sap.ui.webc.base.types.Float}
-		 * @defaultvalue 0.7
-		 * @since 1.0.0-rc.13
-		 */
-		stepSwitchThreshold: {
-			type: Float,
-			defaultValue: STEP_SWITCH_THRESHOLDS.DEFAULT,
-		},
-
-		/**
-		 * Defines the height of the <code>ui5-wizard</code> content.
-		 * @private
-		 */
-		contentHeight: {
-			type: Float,
-		},
-
-		_groupedTabs: {
-			type: String,
-			multiple: true,
-		},
-
-		_breakpoint: {
-			type: String,
-		},
+@event("change", {
+	detail: {
+		step: { type: WizardStep },
```

Comment:
Here you should use previous types. For custom evnts with parameters we introduce WizardChangeEventDetails type which describes them more detailed (like step: WizardStep).

We use event decorator after class description.